internal/version/version.go

```diff
@@ -139,20 +140,64 @@ func parseVersionConstant(versionString, gitCommit string) (*semver.Version, err
 	return &v, nil
 }
 
-func Get() *Info {
+func Get() (*Info, error) {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return nil, errors.New("unable to retrieve build info")
+	}
+
+	const unknown = "unknown"
+	gitCommit := unknown
+	gitTreeState := "clean"
+	gitCommitDate := unknown
+	buildTags := []string{}
+	ldFlags := unknown
+
+	for _, s := range info.Settings {
+		switch s.Key {
+		case "vcs.revision":
+			gitCommit = s.Value
+
+		case "vcs.modified":
+			if s.Value == "true" {
+				gitTreeState = "dirty"
+			}
+
+		case "vcs.time":
+			gitCommitDate = s.Value
+
+		case "-tags":
+			buildTags = strings.Split(s.Value, ",")
+
+		case "-ldflags":
+			ldFlags = s.Value
+		}
+	}
+
+	dependencies := []string{}
```

Comment:
can we add a verbose flag for this? I think it's quite a bit of new output

Reply:
Done :+1: